tinygrad/features/search.py

```diff
@@ -24,6 +24,8 @@ def time_linearizer(lin:Linearizer, rawbufs:List[RawBuffer], allow_test_size=Tru
   key = {"ast": str(lin.ast), "opts": str(lin.applied_opts), "allow_test_size": allow_test_size, "max_global_size": max_global_size}
   if should_copy and not disable_cache and (val:=diskcache_get("time_linearizer", key)) is not None: return min(val)
   if should_copy: lin = lin.copy() # TODO: remove the need for this
+  # allocate a scratch buffer if output buffer is also input
+  if rawbufs[0] in rawbufs[1:]: rawbufs = [cast(Compiled, Device[Device.DEFAULT]).buffer(rawbufs[0].size, rawbufs[0].dtype), *rawbufs[1:]]
```

Comment:
Hmm, maybe this should be outside the search, otherwise it's allocing and deallocating this buffer a lot.

Reply:
okay let me find a way to balance the design and the speed. initially i put it here as we don't expect a timing function to mutate the input.